### PR TITLE
serve list node at once

### PR DESCRIPTION
**Pull request by vinay-debug**
### Related issue
  - *Issue Id*
  - *Context*
  
### List of changes
  - *change a*
  - *change b*

### Database migration/schema changes (if any)
- *List your migrations here...*

### Self review checklist
  - [ ] Confirm that the changes made by you meet the resolution criteria and nothing else.
  - [ ] Ran all the tests locally to check that you have not introduced any new regression.
  - [ ] If there are any failing tests, you have already fixed them. If not 
  - [ ] Followed coding and styling guidelines.
  - [ ] Made sure the changes are is not overly complex, unreadable and unnecessarily smart. Simplicity wins!
  - [ ] Made sure the comments are not too verbose and not to cryptic.
  - [ ] Made sure the code metrics are not deteriorated.
  - [ ] Made sure you have updated (if necessary)-
    - [ ] Documentation
    - [ ] Translation
    - [ ] Tests/test data
    - [ ] Postman collection
    - [ ] Version
       
### Additional info (if any)
- *Add anything you feel worth mentioning...*

### screensshots (if any)

Original | Updated
:------------------------:|-------------------------------|
** Original screenshot ** | ** Updated screenshot **|

### CODE REVIEW

File: src/api/controllers/clinical/assessment/assessment.controller.ts [hunk 1]
```diff
@@ -277,6 +281,80 @@ export class AssessmentController extends BaseController{
...
+            
+            await this.setContext('Assessment.AnswerQuestion', request, response);
+
+            const id: uuid = await this._validator.getParamUuid(request, 'id');
+            const questionIds = request.params.listQuestionId;
+            const questionIdList =  questionIds.split(',');
+            let question = null;
+            let answerResponse: AssessmentQuestionResponseDto = null;
+            const answerResponseList = [];
+
```
Comment (kiran-rean) [inline src/api/controllers/clinical/assessment/assessment.controller.ts:295]: Add check for ServeListNodeChildrenAtOnce to be true before you start answering questions.

File: src/api/controllers/clinical/assessment/assessment.controller.ts [hunk 2]
```diff
@@ -277,6 +281,80 @@ export class AssessmentController extends BaseController{
         }
     };
 
+    answerQuestionList = async (request: express.Request, response: express.Response): Promise<void> => {
+        try {
+            
+            await this.setContext('Assessment.AnswerQuestion', request, response);
+
```
Comment (kiran-rean) [inline src/api/controllers/clinical/assessment/assessment.controller.ts:288]: Add check whether all answers are sent at the start only.

File: src/database/sql/sequelize/repositories/clinical/assessment/assessment.helper.repo.ts [hunk 3]
```diff
@@ -633,19 +633,27 @@ export class AssessmentHelperRepo implements IAssessmentHelperRepo {
...
+                Description                 : nodeObj.Description,
+                Sequence                    : nodeObj.Sequence,
+                Score                       : nodeObj.Score,
+                QueryResponseType           : QueryResponseType.None,
+                ServeListNodeChildrenAtOnce : false
             };
 
+            if (nodeObj.ChildrenNodeDisplayCodes !== undefined) {
+                for await (var childDisplayCode of nodeObj.ChildrenNodeDisplayCodes) {
+                    if (childDisplayCode.startsWith('QNode#') && !nodeObj.DisplayCode.startsWith('RNode#')) {
```
Comment (kiran-rean) [inline src/database/sql/sequelize/repositories/clinical/assessment/assessment.helper.repo.ts:652]:
```
                    if (childDisplayCode.startsWith('QNode#') 
                      && !nodeObj.DisplayCode.startsWith('RNode#')
                     && nodeObj.ServeListNodeChildrenAtOnce  === true) {
                        nodeEntity.ServeListNodeChildrenAtOnce = true;
                    }
```

File: src/domain.types/clinical/assessment/assessment.types.ts [hunk 4]
```diff
@@ -213,13 +215,15 @@ export class CAssessmentListNode extends CAssessmentNode {
...
     Children?               : CAssessmentNode[];
+    ServeListNodeChildrenAtOnce?: boolean;
 
     constructor() {
         super();
         this.NodeType = AssessmentNodeType.NodeList;
         this.ChildrenNodeDisplayCodes = [];
         this.ChildrenNodeIds = [];
         this.Children = [];
+        this.ServeListNodeChildrenAtOnce = null;
```
Comment (kiran-rean) [inline src/domain.types/clinical/assessment/assessment.types.ts:226]: make it false by default. Not as null.

**Review by kiran-rean — CHANGES_REQUESTED** [review]

File: src/api/controllers/clinical/assessment/assessment.controller.ts [hunk 5]
```diff
@@ -277,6 +281,80 @@ export class AssessmentController extends BaseController{
...
+            const id: uuid = await this._validator.getParamUuid(request, 'id');
+            const questionIds = request.params.listQuestionId;
```
[the remark this answers is not in the document]
Comment (vinay-debug) [reply src/api/controllers/clinical/assessment/assessment.controller.ts:290]: changed the context and variable questionIds

File: src/api/controllers/clinical/assessment/assessment.controller.ts (hunk 1 above)
Comment (vinay-debug) [reply src/api/controllers/clinical/assessment/assessment.controller.ts:295]: done

File: src/api/controllers/clinical/assessment/assessment.controller.ts (hunk 2 above)
Comment (vinay-debug) [reply src/api/controllers/clinical/assessment/assessment.controller.ts:288]: done

File: src/database/sql/sequelize/repositories/clinical/assessment/assessment.helper.repo.ts (hunk 3 above)
Comment (vinay-debug) [reply src/database/sql/sequelize/repositories/clinical/assessment/assessment.helper.repo.ts:653]: Third condition not added

**Review by tushar-tikme-co — APPROVED** [review]

**Review by pradnya-tikme — APPROVED** [review]

**Review by pradnya-tikme — APPROVED** [review]

**Review by pradnya-tikme — APPROVED** [review]

**Review by pradnya-tikme — APPROVED** [review]

**Review by kiran-rean — APPROVED** [review]

**Pull request merged**